extended/src/main/java/io/kubernetes/client/extended/controller/ControllerBuilder.java

```diff
@@ -0,0 +1,260 @@
+   * @param <ApiListType> the type parameter
+   * @param apiTypeClass the api type class
+   * @return the controller builder . controller watch builder
+   */
+  public <ApiType, ApiListType>
+      ControllerBuilder.ControllerWatchBuilder<ApiType, ApiListType> watch(
+          Class<ApiType> apiTypeClass) {
+    return new ControllerBuilder.ControllerWatchBuilder<>(apiTypeClass);
+  }
+
+  /**
+   * Overrides name for the controller.
+   *
+   * @param controllerName the controller name
+   * @return the controller builder
+   */
+  public ControllerBuilder withName(String controllerName) {
+    this.controllerName = controllerName;
+    return this;
+  }
+
+  /**
+   * Overrides workQueue for the controller.
+   *
+   * @param workQueue the work queue
+   * @return the controller builder
+   */
+  public ControllerBuilder withWorkQueue(RateLimitingQueue<Reconciler.Request> workQueue) {
+    this.workQueue = workQueue;
+    return this;
+  }
+
+  /**
+   * Add a ready-function to the pre-flight check of the controller.
+   *
+   * @param readyFunc the ready func
+   * @return the controller builder
+   */
+  public ControllerBuilder withReadyFunc(Supplier<Boolean> readyFunc) {
+    this.readyFuncs.add(readyFunc);
+    return this;
+  }
+
+  /**
+   * Use a leader-elector to manage the life-cycle of the controller.
+   *
+   * @param leaderElector the leader elector
+   * @return the controller builder
+   */
+  public ControllerBuilder withLeaderElector(LeaderElector leaderElector) {
+    this.leaderElector = leaderElector;
+    return this;
+  }
+
+  /**
+   * Overrides worker thread counts of the controller.
+   *
+   * @param workerCount the worker count
+   * @return the controller builder
+   */
+  public ControllerBuilder withWorkerCount(int workerCount) {
+    this.workerCount = workerCount;
+    return this;
+  }
+
+  /**
+   * Sets reconciler of the controller.
+   *
+   * @param reconciler the reconciler
+   * @return the controller builder
+   */
+  public ControllerBuilder withReconciler(Reconciler reconciler) {
+    this.reconciler = reconciler;
+    return this;
+  }
+
+  /**
+   * Build the controller.
+   *
+   * @return the controller
+   * @throws IllegalStateException the illegal state exception
+   */
+  public Controller build() throws IllegalStateException {
+
+    DefaultController controller =
+        new DefaultController(
+            this.reconciler, this.workQueue, this.readyFuncs.stream().toArray(Supplier[]::new));
+
+    controller.setName(this.controllerName);
+    controller.setWorkerCount(this.workerCount);
+    controller.setWorkerThreadPool(
+        Executors.newScheduledThreadPool(
+            this.workerCount, Controllers.namedControllerThreadFactory(this.controllerName)));
+
+    if (reconciler == null) {
+      throw new IllegalStateException("Missing reconciler when building controller.");
+    }
+    controller.setReconciler(reconciler);
+
+    if (leaderElector != null) {
```

Comment:
this.leaderElector (consistency with the call below)

Reply:
done